File: src/backend/SyllabusManager.Data/Models/Subjects/CardEntries.cs
using SyllabusManager.Data.Enums.Subjects;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SyllabusManager.Data.Models.Subjects
{
    public class CardEntries : ModelBase
    {
        [Required]
        public string Name { get; set; }
        public SubjectCardEntryType Type { get; set; }
        public List<CardEntry> Entries { get; set; }
    }
}

File: src/frontend/src/app/core/models/subject/card-entries.ts
import { SubjectCardEntryType } from "../../enums/subject/subject-card-entry-type.enum";
import { ModelBase } from "../model-base";
import { CardEntry } from "./card-entry";

export class CardEntries extends ModelBase {
    name: string = '';
    type: SubjectCardEntryType = SubjectCardEntryType.Goal;
    entries: CardEntry[] = [];
}
